superset/assets/javascripts/chart/chartAction.js

```diff
@@ -107,15 +108,27 @@ export const RUN_QUERY = 'RUN_QUERY';
 export function runQuery(formData, force = false, timeout = 60, key) {
   return (dispatch) => {
     const url = getExploreUrl(formData, 'json', force);
+    let logStart;
     const queryRequest = $.ajax({
       url,
       dataType: 'json',
       timeout: timeout * 1000,
+      beforeSend: () => {
+        logStart = Logger.getTimestamp();
+      },
     });
 
     const queryPromise = Promise.resolve(dispatch(chartUpdateStarted(queryRequest, key)))
       .then(() => queryRequest)
-      .then(queryResponse => dispatch(chartUpdateSucceeded(queryResponse, key)))
+      .then((queryResponse) => {
+        Logger.append(LOAD_EVENT, {
+          label: key,
+          isCached: !force,
+          startAt: logStart,
+          duration: Logger.getTimestamp() - logStart,
+        });
+        dispatch(chartUpdateSucceeded(queryResponse, key));
```

Comment:
I think for redux/promise chaining you need to return the `dispatch`? which this function did before. I think it's not broken now but if you were to chain something else that might not work.

Reply:
Yes. should return dispatch.